slice/internal/controller/workload_controller.go

```diff
@@ -172,10 +172,14 @@ func (r *WorkloadReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 		return ctrl.Result{}, err
 	}
 
-	err = r.syncSlices(ctx, wl, ac, &slices, nodes)
+	newSlices, err := r.syncSlices(ctx, wl, ac, slices, nodes)
 	if err != nil {
 		return ctrl.Result{}, err
 	}
+	if len(newSlices) > 0 {
+		slices = append(slices, newSlices...)
+		grouped = r.groupSlices(slices)
+	}
```

Comment:
Can we maybe syncSlices before grouping them in L166 or there's something preventing it. Not saying we must do it, just asking for better understanding

Reply:
SGTM